artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/BindingsImpl.java

```diff
@@ -117,7 +117,9 @@ public void addBinding(final Binding binding) {
             logger.trace("addBinding(" + binding + ") being called");
          }
          if (binding.isExclusive()) {
-            exclusiveBindings.add(binding);
+            if (!exclusiveBindings.contains(binding)) {
```

Comment:
Just because in the else branch non exclusive bindings are using CopyOnWriteArrayList and contains the same validation.
`if (!bindings.contains(binding)) {
   bindings.add(binding);
}`

Reply:
@ldebello good point: changing everything worth only if it makes sense indeed :+1:

Reply:
ok great, so it is ok to let the code as it is or do you suggest to change everything?

Reply:
If we are expecting set like semantics. Then use a set

Reply:
@michaelandrepearce totally agree but that means I should change everything for non exclusive and exclusive binding which I think is overkill to fix this issue based on the fact binding are using the same strategy.

Reply:
The change is just to the collection exclusive bindings. You are making set semantics on that

Reply:
I agree with @michaelandrepearce. I don't think it's overkill to change the collection.

Reply:
@michaelandrepearce @jbertram change the structure from list to set in exclusive binding should be easy but my point is also bindings have set semantics https://github.com/apache/activemq-artemis/blob/65d01a4f9195737708ec82ee216afe3f196b27bc/artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/BindingsImpl.java#L136 so change only one side is a little weird for me and changing everything sounds like doing something different.

Reply:
I looked at this a bit deeper. Order is important for the non-exclusive bindings so it should remain as a `java.util.List`. However, order isn't important for the exclusive bindings so it should either be a `java.util.Set` of some kind or the contains/add on the `java.util.List` should be atomic. Another PR can be sent to deal with the contains/add on the non-exclusive bindings.

Reply:
@jbertram thanks for the information, so I can update exclusive bindings to use `CopyOnWriteArraySet` so in that way we avoid duplicates and warranty being atomic. Is that ok for you?

Reply:
@ldebello, that sounds fine to me.

Reply:
CopyOnArraySet i beleive keeps order as its underpinned/extends by CopyOnArrayList

Reply:
@michaelandrepearce, according to the [JavaDoc](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CopyOnWriteArraySet.html) order is maintained as you note. However, `getNextBinding` indexes into the list so changing to a set would require that method to be refactored substantially. I think it's probably safer just to make the contains/add atomic than to change to a set.

Reply:
@jbertram i agree for scope of this PR simply the changing exclusiveBindings to CopyOnWriteArraySet is enough. 



I was only mearly commenting that behaviour of that specific set implementation. Probably added confusion to this PR.

Reply:
@jbertram sorry for the question, but anyway to re-run the travis job?because it seems it timeouts but I do not see any failing tests

Reply:
Just change commit comment slighty and force push

Reply:
@michaelandrepearce thanks for the tip